File: old_trunk/cost/iona/orbacus/java/obv/TestNode_impl.java
// **********************************************************************
//
// Copyright (c) 2000
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package test.obv;

public class TestNode_impl extends TestNode
{
    public
    TestNode_impl(int n)
    {
        count = n;
    }

    public int
    compute_count()
    {
        int result = count;
        if(left != null)
            result += left.compute_count();
        if(right != null)
            result += right.compute_count();
        return result;
    }
}

File: cost/iona/orbacus/cpp/obv/TestNode_impl.cpp
// **********************************************************************
//
// Copyright (c) 1999
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <TestOBV.h>
#include <TestNode_impl.h>

using namespace CORBA;

//
// IDL:TestNode:1.0
//
TestNode_impl::TestNode_impl(ULong n)
{
    count(n);
}

TestNode_impl::~TestNode_impl()
{
}

#ifdef OMNI_HAVE_COVARIANT_RETURNS
TestNode*
#else
ValueBase*
#endif
TestNode_impl::_copy_value()
{
    TestNode_impl* result = new TestNode_impl(count());
    TestNode* n;
    n = left();
    if(n != 0)
        result -> left(TestNode::_downcast(n -> _copy_value()));
    n = right();
    if(n != 0)
        result -> right(TestNode::_downcast(n -> _copy_value()));
    return result;
}

//
// IDL:TestNode/compute_count:1.0
//
ULong
TestNode_impl::compute_count()
{
    ULong result = count();
    if(left() != 0)
        result += left() -> compute_count();
    if(right() != 0)
        result += right() -> compute_count();
    return result;
}
